platform/viewer/src/appExtensions/MeasurementsPanel/init.js

```diff
@@ -131,15 +45,38 @@ export default function init({ servicesManager, configuration = {} }) {
     this,
     'labelmapModified'
   );
-  //
-  const onRightClick = getOnRightClickCallback(store);
-  const onTouchPress = getOnTouchPressCallback(store);
-  const onNewImage = getResetLabellingAndContextMenu(store);
-  const onMouseClick = getResetLabellingAndContextMenu(store);
-  const onTouchStart = getResetLabellingAndContextMenu(store);
-
-  // Because click gives us the native "mouse up", buttons will always be `0`
-  // Need to fallback to event.which;
+
+  const onRightClick = event => {
+    UIContextMenuService.show({ event: event.detail });
+  };
+
+  const onTouchPress = event => {
+    UIContextMenuService.show({
+      event: event.detail,
+      props: {
+        isTouchEvent: true,
+      },
+    });
+  };
+
+  const onTouchStart = () => {
+    UILabellingFlowService.hide();
+    UIContextMenuService.hide();
+  };
+
+  const onMouseClick = () => {
+    UIContextMenuService.hide();
+    UILabellingFlowService.hide();
+  };
```

Comment:
We have no guarantees that a service is registered; we need to check if they are before calling methods on them.

We should probably create follow-up issues to make a more generic contextmenu? Next step would be this extension provided context options + actions when calling `show`.

Similarly, labeling flow, in the future, could probably be a DialogService with DialogContent passed from this extension?

Reply:
Alright, done. Added in the comment about (wontfix) this follow-up issue.